src/ghost-content/index.js

```diff
@@ -192,10 +196,29 @@ module.exports = ({ logger, db, projectLocation }) => {
 
   const getPending = () => pendingRevisionsByFolder
 
+  const getPendingWithContentForFolder = async (folderInfo, normalizedFolderName) => {
+    const revisions = folderInfo.map(({ revision }) => revision)
+    const fileNames = uniq(folderInfo.map(({ file }) => file))
+
+    const knex = await db.get()
+    const files = await knex('ghost_content')
+      .select('file', 'content')
+      .whereIn('file', fileNames)
+      .andWhere({ folder: normalizedFolderName })
```

Comment:
That should work for newly created files within a specific folder but what about files created in subfolders? For example Flows can have subfolders. e.g.

```
/flows
  a.flow.json
  a.ui.json
  /skills
    b.flow.json
    b.ui.json
```

Should we make it `andWhere('folder', 'like', normalizedFolderName + '%')` ?

Reply:
It will work. By design in the ghost manager you register an _entry point_ and glob, like `flows` and `**/*.json`. 
That starts ghost-tracking the entire folder and every glob-matching file inside of it.
They will be recorded as `(flows, a.flow.json), (flows, a.ui.json), (flows, skills/b.flow.json), (flow, skills/b.ui.json)`.
The folder serves as type of the content (content, flows, nlp, anything else). It will have a single `.ghost-revisions` file inside.